internal/saucecloud/artifactservice.go

```diff
@@ -0,0 +1,80 @@
+package saucecloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/saucelabs/saucectl/internal/artifacts"
+	"github.com/saucelabs/saucectl/internal/rdc"
+	"github.com/saucelabs/saucectl/internal/resto"
+	"github.com/saucelabs/saucectl/internal/testcomposer"
+)
+
+// ArtifactService represents artifact service
+type ArtifactService struct {
+	JobService
+}
+
+// NewArtifactService returns an artifact service
+func NewArtifactService(restoClient resto.Client, rdcClient rdc.Client, testcompClient testcomposer.Client) *ArtifactService {
+	return &ArtifactService{
+		JobService: JobService{
+			VDCReader: &restoClient,
+			RDCReader: &rdcClient,
+			VDCWriter: &testcompClient,
+		},
+	}
+}
+
+// List returns an artifact list
+func (s *ArtifactService) List(jobID string) (artifacts.List, error) {
+	isRDC, err := s.isRDC(jobID)
+	if err != nil {
+		return artifacts.List{}, err
+	}
+	items, err := s.GetJobAssetFileNames(context.Background(), jobID, isRDC)
+	if err != nil {
+		return artifacts.List{}, err
+	}
+	return artifacts.List{
+		JobID: jobID,
+		Items: items,
+	}, nil
+}
+
+// Download does download specified artifacts
+func (s *ArtifactService) Download(jobID, filename string) ([]byte, error) {
+	isRDC, err := s.isRDC(jobID)
+	if err != nil {
+		return []byte{}, err
```

Comment:
It's OK to return `nil` in that case 👍 

```suggestion
		return nil, err
```